OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -1165,4 +1140,77 @@ await UpdateStatus(new()
             }).ConfigureAwait(false);
         }
     }
+
+    private async Task<Workshop> CheckDtoAndPrepareCreatedWorkshop(WorkshopCreateUpdateDto dto)
+    {
+        if (dto.MemberOfWorkshopId.HasValue && !await Exists((Guid)dto.MemberOfWorkshopId).ConfigureAwait(false))
```

Comment:
MemberOfWorkshopId property of WorkshopBaseDto class has to be 'Guid?' type. Because of created Workshop may not be a member of other Workshop.
The Exists(Guid id) method has an argument of type 'Guid', not 'Guid?'. I don't want to overload or change this method, so I cast 'Guid?' to 'Guid'.
It's cast safely because I check dto.MemberOfWorkshopId.HasValue before doing so.